docs/resources/notification_destination.md

```diff
@@ -0,0 +1,102 @@
+    config {
+        email {
+            addresses = ["abc@gmail.com"]
+        }
+    }
+}
+```
+`Slack` notification destination:
+
+```hcl
+resource "databricks_notification_destination" "ndresource" {
+    display_name = "Notification Destination"
+    config {
+        slack {
+            url     = "https://hooks.slack.com/services/..."
+        }
+    }
+}
+```
+`PagerDuty` notification destination:
+
+```hcl
+resource "databricks_notification_destination" "ndresource" {
+    display_name = "Notification Destination"
+    config {
+        pagerduty {
+            integration_key = "xxxxxx"
+        }
+    }
+}
+```
+`Microsoft Teams` notification destination:
+
+```hcl
+resource "databricks_notification_destination" "ndresource" {
+    display_name = "Notification Destination"
+    config {
+        microsoft_teams {
+            url = "https://outlook.office.com/webhook/..."
+        }
+    }
+}
+```
+`Generic Webhook` notification destination:
+
+```hcl
+resource "databricks_notification_destination" "ndresource" {
+    display_name = "Notification Destination"
+    config {
+        generic_webhook {
+            url = "https://example.com/webhook"
+            username = "username" // Optional
+            password = "password" // Optional
+        }
+    }
+}
+```
+
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `display_name` - (Required) The display name of the Notification Destination.
+* `config` - (Required) The configuration of the Notification Destination. It must contain exactly one of the following blocks:
+  * `email` - The email configuration of the Notification Destination. It must contain the following:
+    * `addresses` - (Required) The list of email addresses to send notifications to.
+  * `slack` - The Slack configuration of the Notification Destination. It must contain the following:
+    * `url` - (Required) The Slack webhook URL.
+  * `pagerduty` - The PagerDuty configuration of the Notification Destination. It must contain the following:
+    * `integration_key` - (Required) The PagerDuty integration key.
+  * `microsoft_teams` - The Microsoft Teams configuration of the Notification Destination. It must contain the following:
+    * `url` - (Required) The Microsoft Teams webhook URL.
+  * `generic_webhook` - The Generic Webhook configuration of the Notification Destination. It must contain the following:
+    * `url` - (Required) The Generic Webhook URL.
+    * `username` - (Optional) The username for basic authentication.
+    * `password` - (Optional) The password for basic authentication.
+
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The unique ID of the Notification Destination.
+
+
+## Note
+* If the type of notification destination is changed, the existing notification destination will be deleted and a new notification destination will be created with the new type.
```

Comment:
It's better to move this note close to the arguments reference, and use this syntax to render it as a block:

```
-> **Note** If the type of notification destination is changed ...
```